Terminal not executing commands entered before prompt is printed

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

The terminal is not executing commands entered before the prompt is printed when Oh My Posh is used. When not using Oh My Posh, the terminal executes commands entered before a prompt is printed, producing expected behavior.  
  
<details>
<summary>Actual behavior</summary>
  
<img width="1280" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/45308169/5e99992a-f912-42e7-8d48-a3deedbd7fd8">
  
<img width="1280" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/45308169/66955b51-d337-4928-9634-57b0a124b173" />

</details>
  
<details>
<summary>Expected behavior</summary>
  
<img width="1280" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/45308169/50f9db35-8da3-42bb-9902-3dc22eaa8943">
  
<img width="1280" alt="image" src="https://github.com/JanDeDobbeleer/oh-my-posh/assets/45308169/c86defd1-944a-4a97-ba6e-cece8b42f44f">

</details>

### Theme

Custom theme:  
  
<details>
<summary>JSON</summary>

```json
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "version": 2,
  "final_space": true,
  "terminal_background": "transparent",
  "blocks": [
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "type": "os",
          "style": "diamond",
          "foreground": "p:white",
          "background": "p:black",
          "leading_diamond": "",
          "trailing_diamond": "",
          "properties": {
            "ubuntu": "<p:theme></> ",
            "windows": "<p:theme></> "
          }
        },
        {
          "type": "shell",
          "style": "diamond",
          "foreground": "p:white",
          "background": "p:black",
          "leading_diamond": "",
          "trailing_diamond": ""
        },
        {
          "type": "sysinfo",
          "style": "diamond",
          "foreground": "p:white",
          "background": "p:black",
          "leading_diamond": "",
          "trailing_diamond": "",
          "template": "<p:theme></> CPU: {{ round .PhysicalPercentUsed .Precision }}% | RAM: {{ (div ((sub .PhysicalTotalMemory .PhysicalFreeMemory)|float64) 1000000000.0) }}/{{ (div .PhysicalTotalMemory 1000000000.0) }}GB <p:theme></> "
        }
      ],
      "newline": true
    },
    {
      "type": "prompt",
      "alignment": "right",
      "segments": [
        {
          "type": "executiontime",
          "style": "diamond",
          "foreground": "p:white",
          "background": "p:black",
          "leading_diamond": "",
          "trailing_diamond": "",
          "template": "<p:theme></> {{ .FormattedMs }} <p:theme></> ",
          "properties": {
            "style": "houston",
            "threshold": 0
          }
        },
        {
          "type": "time",
          "style": "diamond",
          "foreground": "p:white",
          "background": "p:black",
          "leading_diamond": "",
          "trailing_diamond": "",
          "properties": {
            "time_format": "<p:theme></> Monday, 2006-01-02 15:04:05.000 <p:theme></> "
          }
        }
      ]
    },
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "type": "session",
          "style": "diamond",
          "foreground": "p:theme",
          "background": "p:black",
          "leading_diamond": "",
          "trailing_diamond": ""
        },
        {
          "type": "path",
          "style": "diamond",
          "foreground": "p:black",
          "background": "p:theme",
          "leading_diamond": "",
          "trailing_diamond": "",
          "properties": {
            "style": "full"
          }
        }
      ],
      "newline": true
    },
    {
      "type": "prompt",
      "alignment": "right",
      "segments": [
        {
          "type": "git",
          "style": "diamond",
          "foreground": "p:black",
          "background": "p:green",
          "leading_diamond": "",
          "trailing_diamond": "",
          "template": " {{ .UpstreamIcon }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }}  {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }}  {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }}  {{ .StashCount }}{{ end }} ",
          "properties": {
            "branch_icon": " ",
            "fetch_stash_count": true,
            "fetch_status": true,
            "fetch_upstream_icon": true,
            "fetch_worktree_count": true
          }
        }
      ]
    },
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "type": "text",
          "style": "plain",
          "template": "<p:theme>$</>"
        }
      ],
      "newline": true
    }
  ],
  "palettes": {
    "template": "{{ if .WSL }}ubuntu{{ else }}windows{{ end }}",
    "list": {
      "windows": {
        "black": "#000000",
        "blue": "#4CC2FF",
        "green": "#1DB954",
        "orange": "#F39456",
        "white": "#FFFFFF",
        "theme": "p:blue"
      },
      "ubuntu": {
        "black": "#000000",
        "blue": "#4CC2FF",
        "green": "#1DB954",
        "orange": "#F39456",
        "white": "#FFFFFF",
        "theme": "p:orange"
      }
    }
  }
}
```

</details>

### What OS are you seeing the problem on?

Windows, Linux

### Which shell are you using?

bash

### Log output
  
<details>
<summary>Logs</summary>

```Shell
Version: 17.5.2

Shell: bash (5.1.16(1)-release)

Segments:

ConsoleTitle(false)                        -   0 ms - 
os(true)                                   -   0 ms -  WSL at   
shell(true)                                -   0 ms -  bash 
sysinfo(true)                              -   0 ms -  CPU: 34.84% | RAM: 4/8GB  
executiontime(true)                        -   1 ms -  00:00:00.0  
time(true)                                 -   0 ms -   Thursday, 2023-07-20 02:49:53.329   
session(true)                              -   0 ms -  sfm61319@Spidey 
path(true)                                 -   0 ms -  ~ 
git(false)                                 -   0 ms - 
text(true)                                 -   0 ms - $

Run duration: 12.512528ms

Cache path: /home/sfm61319/.cache/oh-my-posh

Config path: /mnt/c/Users/Svass/AppData/Local/Programs/oh-my-posh/themes/sigma.omp.json

Logs:

[DEBUG] 02:49:53.319 shell.go:Getenv:398 → NO DATA
[TRACE] 02:49:53.319 shell.go:Getenv(TERM_PROGRAM) - 17.202µs
[DEBUG] 02:49:53.319 shell.go:Getenv:398 → NO DATA
[TRACE] 02:49:53.319 shell.go:Getenv(XDG_CACHE_HOME) - 2.385µs
[TRACE] 02:49:53.319 shell_unix.go:CachePath() - 12.414µs
[DEBUG] 02:49:53.319 shell.go:Getenv:398 → /mnt/c/Users/Svass/AppData/Local/Programs/oh-my-posh/themes/sigma.omp.json
[TRACE] 02:49:53.319 shell.go:Getenv(POSH_THEME) - 4.258µs
[DEBUG] 02:49:53.319 shell_unix.go:Platform:71 → ubuntu
[TRACE] 02:49:53.319 shell.go:resolveConfigPath() - 13.215µs
[TRACE] 02:49:53.319 shell.go:Init() - 178.198µs
[TRACE] 02:49:53.319 shell.go:Flags() - 40ns
[TRACE] 02:49:53.325 config.go:loadConfig() - 6.625691ms
[TRACE] 02:49:53.325 shell.go:Flags() - 120ns
[DEBUG] 02:49:53.325 shell.go:Getenv:398 → NO DATA
[TRACE] 02:49:53.325 shell.go:Getenv(OMP_CACHE_DISABLED) - 5.14µs
[TRACE] 02:49:53.326 shell_unix.go:Root() - 672ns
[DEBUG] 02:49:53.326 shell.go:Shell:645 → process name: bash
[TRACE] 02:49:53.326 shell.go:Shell() - 465.925µs
[TRACE] 02:49:53.326 shell.go:ErrorCode() - 100ns
[DEBUG] 02:49:53.326 shell_unix.go:IsWsl:35 → true
[TRACE] 02:49:53.326 shell_unix.go:IsWsl() - 3.657µs
[TRACE] 02:49:53.326 shell.go:GOOS() - 60ns
[DEBUG] 02:49:53.326 shell.go:Pwd:428 → /home/sfm61319
[TRACE] 02:49:53.326 shell.go:Pwd() - 10.249µs
[TRACE] 02:49:53.326 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 20ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 20ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 20ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 20ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 21ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 50ns
[DEBUG] 02:49:53.326 shell.go:User:549 → sfm61319
[TRACE] 02:49:53.326 shell.go:User() - 6.883µs
[DEBUG] 02:49:53.326 shell.go:Host:561 → Spidey
[TRACE] 02:49:53.326 shell.go:Host() - 3.346µs
[TRACE] 02:49:53.326 shell.go:GOOS() - 30ns
[DEBUG] 02:49:53.326 shell_unix.go:Platform:71 → ubuntu
[DEBUG] 02:49:53.326 shell.go:Getenv:398 → 1
[TRACE] 02:49:53.326 shell.go:Getenv(SHLVL) - 2.084µs
[TRACE] 02:49:53.326 shell.go:TemplateCache() - 553.641µs
[TRACE] 02:49:53.326 shell.go:Shell() - 60ns
[DEBUG] 02:49:53.326 shell.go:Getenv:398 → 5.1.16(1)-release
[TRACE] 02:49:53.326 shell.go:Getenv(POSH_SHELL_VERSION) - 2.444µs
[DEBUG] 02:49:53.326 debug.go:PrintDebug:23 → Segment: Title
[TRACE] 02:49:53.326 shell.go:Flags() - 30ns
[DEBUG] 02:49:53.326 block.go:Debug:280 → Segment: os
[TRACE] 02:49:53.326 shell.go:Pwd() - 60ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.326 shell.go:GOOS() - 30ns
[DEBUG] 02:49:53.326 shell_unix.go:Platform:71 → ubuntu
[DEBUG] 02:49:53.326 properties.go:GetBool:22 → display_distro_name: false
[DEBUG] 02:49:53.326 properties.go:GetString:28 → <p:theme></> 
[TRACE] 02:49:53.326 shell.go:TemplateCache() - 100ns
[TRACE] 02:49:53.326 shell.go:TemplateCache() - 211ns
[TRACE] 02:49:53.326 shell.go:Shell() - 151ns
[DEBUG] 02:49:53.327 block.go:Debug:280 → Segment: shell
[TRACE] 02:49:53.327 shell.go:Pwd() - 110ns
[TRACE] 02:49:53.327 shell.go:GOOS() - 40ns
[DEBUG] 02:49:53.327 properties.go:GetKeyValueMap:46 → mapped_shell_names: map[]
[TRACE] 02:49:53.327 shell.go:Shell() - 40ns
[TRACE] 02:49:53.327 shell.go:Flags() - 30ns
[TRACE] 02:49:53.327 shell.go:TemplateCache() - 100ns
[TRACE] 02:49:53.327 shell.go:TemplateCache() - 80ns
[TRACE] 02:49:53.327 shell.go:Shell() - 50ns
[DEBUG] 02:49:53.327 block.go:Debug:280 → Segment: sysinfo
[DEBUG] 02:49:53.327 properties.go:GetInt:40 → precision: 2
[TRACE] 02:49:53.327 shell.go:Pwd() - 100ns
[TRACE] 02:49:53.327 shell.go:GOOS() - 31ns
[TRACE] 02:49:53.327 shell.go:TemplateCache() - 50ns
[TRACE] 02:49:53.327 shell.go:TemplateCache() - 70ns
[TRACE] 02:49:53.327 shell.go:Shell() - 100ns
[TRACE] 02:49:53.327 shell.go:Flags() - 30ns
[DEBUG] 02:49:53.327 block.go:Debug:280 → Segment: executiontime
[TRACE] 02:49:53.327 shell.go:Pwd() - 70ns
[TRACE] 02:49:53.327 shell.go:GOOS() - 30ns
[DEBUG] 02:49:53.327 properties.go:GetBool:22 → always_enabled: false
[TRACE] 02:49:53.327 shell.go:ExecutionTime() - 40ns
[DEBUG] 02:49:53.327 properties.go:GetFloat64:34 → threshold: 0.000000
[DEBUG] 02:49:53.327 properties.go:GetString:28 → houston
[TRACE] 02:49:53.327 shell.go:TemplateCache() - 120ns
[TRACE] 02:49:53.329 shell.go:TemplateCache() - 321ns
[TRACE] 02:49:53.329 shell.go:Shell() - 100ns
[DEBUG] 02:49:53.329 block.go:Debug:280 → Segment: time
[TRACE] 02:49:53.329 shell.go:Pwd() - 80ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 30ns
[DEBUG] 02:49:53.329 properties.go:GetString:28 → <p:theme></> Monday, 2006-01-02 15:04:05.000 <p:theme></> 
[TRACE] 02:49:53.329 shell.go:TemplateCache() - 100ns
[TRACE] 02:49:53.329 shell.go:TemplateCache() - 81ns
[TRACE] 02:49:53.329 shell.go:Shell() - 50ns
[TRACE] 02:49:53.329 shell.go:Flags() - 30ns
[DEBUG] 02:49:53.329 block.go:Debug:280 → Segment: session
[TRACE] 02:49:53.329 shell.go:Pwd() - 40ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 40ns
[DEBUG] 02:49:53.329 shell.go:Getenv:398 → NO DATA
[TRACE] 02:49:53.329 shell.go:Getenv(SSH_CONNECTION) - 2.355µs
[DEBUG] 02:49:53.329 shell.go:Getenv:398 → NO DATA
[TRACE] 02:49:53.329 shell.go:Getenv(SSH_CLIENT) - 1.403µs
[TRACE] 02:49:53.329 shell.go:TemplateCache() - 60ns
[TRACE] 02:49:53.329 shell.go:TemplateCache() - 90ns
[TRACE] 02:49:53.329 shell.go:Shell() - 60ns
[DEBUG] 02:49:53.329 block.go:Debug:280 → Segment: path
[TRACE] 02:49:53.329 shell.go:Pwd() - 30ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.329 shell.go:Pwd() - 30ns
[TRACE] 02:49:53.329 shell.go:Shell() - 20ns
[TRACE] 02:49:53.329 shell.go:Shell() - 20ns
[DEBUG] 02:49:53.329 properties.go:GetBool:22 → mapped_locations_enabled: true
[DEBUG] 02:49:53.329 properties.go:GetString:28 → 
[DEBUG] 02:49:53.329 properties.go:GetString:28 → 
[TRACE] 02:49:53.329 shell.go:GOOS() - 20ns
[DEBUG] 02:49:53.329 properties.go:GetString:28 → ~
[DEBUG] 02:49:53.329 properties.go:GetKeyValueMap:46 → mapped_locations: map[]
[TRACE] 02:49:53.329 shell.go:PathSeparator() - 40ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.329 shell.go:PathSeparator() - 20ns
[TRACE] 02:49:53.329 shell.go:PathSeparator() - 70ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 20ns
[TRACE] 02:49:53.329 shell.go:PathSeparator() - 30ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 20ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 30ns
[TRACE] 02:49:53.329 shell.go:PathSeparator() - 30ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 20ns
[TRACE] 02:49:53.329 shell.go:GOOS() - 20ns
[TRACE] 02:49:53.329 shell.go:PathSeparator() - 20ns
[TRACE] 02:49:53.330 shell.go:Pwd() - 30ns
[DEBUG] 02:49:53.330 shell_unix.go:IsWsl:35 → true
[TRACE] 02:49:53.330 shell_unix.go:IsWsl() - 2.635µs
[DEBUG] 02:49:53.330 shell.go:RunCommand:579 → //wsl.localhost/Ubuntu/home/sfm61319
[TRACE] 02:49:53.330 shell.go:RunCommand(wslpath -m /home/sfm61319) - 663.79µs
[TRACE] 02:49:53.330 shell.go:StackCount() - 80ns
[TRACE] 02:49:53.330 shell_unix.go:DirIsWritable(/home/sfm61319) - 4.158µs
[TRACE] 02:49:53.330 shell.go:TemplateCache() - 171ns
[TRACE] 02:49:53.330 shell.go:TemplateCache() - 181ns
[TRACE] 02:49:53.330 shell.go:Shell() - 90ns
[TRACE] 02:49:53.330 shell.go:Flags() - 30ns
[DEBUG] 02:49:53.330 block.go:Debug:280 → Segment: git
[TRACE] 02:49:53.330 shell.go:Pwd() - 60ns
[TRACE] 02:49:53.330 shell.go:GOOS() - 40ns
[DEBUG] 02:49:53.330 shell_unix.go:IsWsl:35 → true
[TRACE] 02:49:53.330 shell_unix.go:IsWsl() - 2.044µs
[DEBUG] 02:49:53.330 shell.go:FileContent:523 ↓
    5.15.90.1-microsoft-standard-WSL2
    
[TRACE] 02:49:53.330 shell.go:FileContent(/proc/sys/kernel/osrelease) - 40.387µs
[TRACE] 02:49:53.330 shell_unix.go:IsWsl2() - 47.01µs
[TRACE] 02:49:53.330 shell.go:Pwd() - 60ns
[DEBUG] 02:49:53.331 shell.go:RunCommand:579 → //wsl.localhost/Ubuntu/home/sfm61319
[TRACE] 02:49:53.331 shell.go:RunCommand(wslpath -m /home/sfm61319) - 453.401µs
[TRACE] 02:49:53.331 shell.go:GOOS() - 50ns
[DEBUG] 02:49:53.331 shell.go:CommandPath:600 → /usr/bin/git
[TRACE] 02:49:53.331 shell.go:CommandPath(git) - 19.437µs
[TRACE] 02:49:53.331 shell.go:HasCommand(git) - 23.675µs
[TRACE] 02:49:53.331 shell.go:Pwd() - 100ns
[ERROR] 02:49:53.331 shell.go:HasParentFilePath:733 → stat .git: no such file or directory
[TRACE] 02:49:53.331 shell.go:HasParentFilePath(.git) - 17.393µs
[DEBUG] 02:49:53.331 properties.go:GetBool:22 → fetch_bare_info: false
[TRACE] 02:49:53.331 shell.go:Flags() - 40ns
[DEBUG] 02:49:53.331 block.go:Debug:280 → Segment: text
[TRACE] 02:49:53.331 shell.go:Pwd() - 40ns
[TRACE] 02:49:53.331 shell.go:GOOS() - 40ns
[TRACE] 02:49:53.331 shell.go:TemplateCache() - 120ns
[TRACE] 02:49:53.331 shell.go:Shell() - 101ns
[DEBUG] 02:49:53.331 shell.go:Getenv:398 → NO DATA
[TRACE] 02:49:53.331 shell.go:Getenv(XDG_CACHE_HOME) - 2.805µs
[TRACE] 02:49:53.331 shell_unix.go:CachePath() - 10.651µs
[TRACE] 02:49:53.331 shell.go:Flags() - 40ns
```

</details>

--- a/scripts/bluesky/main.js
+++ b/scripts/bluesky/main.js
@@ -16,6 +16,7 @@ const {Octokit} = require("@octokit/rest");
   notes = notes.replaceAll('**', '');
   notes = notes.replace(/ \(\[[0-9a-z]+\]\(.*\)/g, '');
   notes = notes.trim();
+  notes = notes.substring(0, 249);
 
   const agent = new BskyAgent({ service: 'https://bsky.social' });
   await agent.login({ identifier: process.env.BLUESKY_IDENTIFIER, password: process.env.BLUESKY_PASSWORD });

--- a/src/cli/init.go
+++ b/src/cli/init.go
@@ -70,8 +70,8 @@ func runInit(shellName string) {
 	shell.ShellIntegration = cfg.ShellIntegration
 	for i, block := range cfg.Blocks {
 		// only fetch cursor position when relevant
-		if i == 0 && block.Newline {
-			shell.Cursor = true
+		if !cfg.DisableCursorPositioning && (i == 0 && block.Newline) {
+			shell.CursorPositioning = true
 		}
 		if block.Type == engine.RPrompt {
 			shell.RPrompt = true

--- a/src/engine/config.go
+++ b/src/engine/config.go
@@ -34,24 +34,25 @@ const (
 
 // Config holds all the theme for rendering the prompt
 type Config struct {
-	Version              int                    `json:"version"`
-	FinalSpace           bool                   `json:"final_space,omitempty"`
-	ConsoleTitleTemplate string                 `json:"console_title_template,omitempty"`
-	TerminalBackground   string                 `json:"terminal_background,omitempty"`
-	AccentColor          string                 `json:"accent_color,omitempty"`
-	Blocks               []*Block               `json:"blocks,omitempty"`
-	Tooltips             []*Segment             `json:"tooltips,omitempty"`
-	TransientPrompt      *Segment               `json:"transient_prompt,omitempty"`
-	ValidLine            *Segment               `json:"valid_line,omitempty"`
-	ErrorLine            *Segment               `json:"error_line,omitempty"`
-	SecondaryPrompt      *Segment               `json:"secondary_prompt,omitempty"`
-	DebugPrompt          *Segment               `json:"debug_prompt,omitempty"`
-	Palette              ansi.Palette           `json:"palette,omitempty"`
-	Palettes             *ansi.Palettes         `json:"palettes,omitempty"`
-	Cycle                ansi.Cycle             `json:"cycle,omitempty"`
-	ShellIntegration     bool                   `json:"shell_integration,omitempty"`
-	PWD                  string                 `json:"pwd,omitempty"`
-	Var                  map[string]interface{} `json:"var,omitempty"`
+	Version                  int                    `json:"version"`
+	FinalSpace               bool                   `json:"final_space,omitempty"`
+	ConsoleTitleTemplate     string                 `json:"console_title_template,omitempty"`
+	TerminalBackground       string                 `json:"terminal_background,omitempty"`
+	AccentColor              string                 `json:"accent_color,omitempty"`
+	Blocks                   []*Block               `json:"blocks,omitempty"`
+	Tooltips                 []*Segment             `json:"tooltips,omitempty"`
+	TransientPrompt          *Segment               `json:"transient_prompt,omitempty"`
+	ValidLine                *Segment               `json:"valid_line,omitempty"`
+	ErrorLine                *Segment               `json:"error_line,omitempty"`
+	SecondaryPrompt          *Segment               `json:"secondary_prompt,omitempty"`
+	DebugPrompt              *Segment               `json:"debug_prompt,omitempty"`
+	Palette                  ansi.Palette           `json:"palette,omitempty"`
+	Palettes                 *ansi.Palettes         `json:"palettes,omitempty"`
+	Cycle                    ansi.Cycle             `json:"cycle,omitempty"`
+	ShellIntegration         bool                   `json:"shell_integration,omitempty"`
+	PWD                      string                 `json:"pwd,omitempty"`
+	Var                      map[string]interface{} `json:"var,omitempty"`
+	DisableCursorPositioning bool                   `json:"disable_cursor_positioning,omitempty"`
 
 	// Deprecated
 	OSC99 bool `json:"osc99,omitempty"`

--- a/src/shell/init.go
+++ b/src/shell/init.go
@@ -46,12 +46,12 @@ const (
 )
 
 var (
-	Transient        bool
-	ErrorLine        bool
-	Tooltips         bool
-	ShellIntegration bool
-	RPrompt          bool
-	Cursor           bool
+	Transient         bool
+	ErrorLine         bool
+	Tooltips          bool
+	ShellIntegration  bool
+	RPrompt           bool
+	CursorPositioning bool
 )
 
 func getExecutablePath(env platform.Environment) (string, error) {
@@ -270,7 +270,7 @@ func PrintInit(env platform.Environment) string {
 		"::TOOLTIPS::", toggleSetting(Tooltips),
 		"::FTCS_MARKS::", toggleSetting(ShellIntegration),
 		"::RPROMPT::", strconv.FormatBool(RPrompt),
-		"::CURSOR::", strconv.FormatBool(Cursor),
+		"::CURSOR::", strconv.FormatBool(CursorPositioning),
 		"::UPGRADE::", strconv.FormatBool(hasNotice),
 		"::UPGRADENOTICE::", notice,
 	).Replace(script)

--- a/themes/schema.json
+++ b/themes/schema.json
@@ -3269,6 +3269,12 @@
       "description": "https://ohmyposh.dev/docs/configuration/overview#general-settings",
       "default": true
     },
+    "disable_cursor_positioning": {
+      "type": "boolean",
+      "title": "Disable Cursor Positioning",
+      "description": "https://ohmyposh.dev/docs/configuration/overview#general-settings",
+      "default": false
+    },
     "shell_integration": {
       "type": "boolean",
       "title": "FTCS command marks for shell integration",

--- a/website/docs/configuration/overview.mdx
+++ b/website/docs/configuration/overview.mdx
@@ -126,14 +126,15 @@ For example, the following is a valid `--config` flag:
 
 ## General Settings
 
-| Name                  | Type             | Description                                                                                                                                                                           |
-| --------------------- | ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| `final_space`         | `boolean`        | when true adds a space at the end of the prompt                                                                                                                                       |
-| `pwd`                 | `string`         | notify terminal of current working directory, values can be `osc99`, `osc7`, or `osc51` depending on your terminal                                                                    |
-| `terminal_background` | `string`         | [color][colors] - terminal background color, set to your terminal's background color when you notice black elements in Windows Terminal or the Visual Studio Code integrated terminal |
-| `accent_color`        | `string`         | [color][colors] - accent color, used as a fallback when the `accent` [color][accent] is not supported                                                                                 |
-| `var`                 | `map[string]any` | config variables to use in [templates][templates]. Can be any value                                                                                                                   |
-| `shell_integration`   | `boolean`        | enable shell integration using FinalTerm's OSC sequences. Works in bash, cmd (Clink v1.14.25+), fish, powershell and zsh                                                              |
+| Name                         | Type             | Description                                                                                                                                                                           |
+| ---------------------------- | ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `final_space`                | `boolean`        | when true adds a space at the end of the prompt                                                                                                                                       |
+| `pwd`                        | `string`         | notify terminal of current working directory, values can be `osc99`, `osc7`, or `osc51` depending on your terminal                                                                    |
+| `terminal_background`        | `string`         | [color][colors] - terminal background color, set to your terminal's background color when you notice black elements in Windows Terminal or the Visual Studio Code integrated terminal |
+| `accent_color`               | `string`         | [color][colors] - accent color, used as a fallback when the `accent` [color][accent] is not supported                                                                                 |
+| `var`                        | `map[string]any` | config variables to use in [templates][templates]. Can be any value                                                                                                                   |
+| `shell_integration`          | `boolean`        | enable shell integration using FinalTerm's OSC sequences. Works in bash, cmd (Clink v1.14.25+), fish, powershell and zsh                                                              |
+| `disable_cursor_positioning` | `boolean`        | disable fetching the cursor position in bash and zsh in case of unwanted side-effects                                                                                                 |
 
 ### JSON Schema Validation
 

